test/state/mpt.cpp

```diff
@@ -62,24 +70,13 @@ class Path
         return {begin(), begin() + size};
     }
 
-    [[nodiscard]] bytes encode(bool extended) const
+    [[nodiscard]] bytes encode(Kind kind) const
     {
-        bytes bs;
         const auto is_even = m_size % 2 == 0;
-        if (is_even)
-            bs.push_back(0x00);
-        else
-            bs.push_back(0x10 | m_nibbles[0]);
-        for (size_t i = is_even ? 0 : 1; i < m_size; ++i)
-        {
-            const auto h = m_nibbles[i++];
-            const auto l = m_nibbles[i];
-            assert(h <= 0x0f);
-            assert(l <= 0x0f);
-            bs.push_back(static_cast<uint8_t>((h << 4) | l));
-        }
-        if (!extended)
-            bs[0] |= 0x20;
+        bytes bs{static_cast<uint8_t>(
+            (is_even ? 0x00 : (0x10 | m_nibbles[0])) | (kind == Kind::leaf ? 0x20 : 0x00))};
+        for (size_t i = is_even ? 0 : 1; i < m_size; i += 2)
```

Comment:
This function got more compressed but harder to read.

Reply:
Noted.

Reply:
Is the new version any good?